src/main.cpp

```diff
@@ -2383,6 +2393,27 @@ void loopLED() {
     }
 }
 
+void loopWATER() {
```

Comment:
1. Then I fear you don't understand how a state machine works. Of course there would have to be state transitions where appropriate.
2. The water level is not the issue here. There can always be noise at the GPIO bin. You want the signal to be stable for a period before you do anything. I don't care if it works on your bench.

Reply:
1. Hope I do. I think we need both: I would like to indicate "water empty" on the display e.g. also during a brew (or during other things like drawing water, steaming), not after the brew. That would need two MachineState, brew-water-full and brew-water-empty with a transition. To avoid this duplication, a separate Water-State is needed in my eyes.
But whent he machine is "locked" due to missing water, we need a MachineState "kWaterEmpty", with a clear indication that brewing is not possible at the moment. I simply did not implement this yet, I will do that once I used it for a while.
2. Currently, nothing is done except for the icon in the display. Will implement the debounce once the state chanbe is implemented.

Reply:
2. Implemented.
For 1 - need to draw the state diagram to make sure it works. Will come later.